### PR TITLE
adds validations for pair creation

### PR DESCRIPTION
**Pull request by alanbsmith**
#### Why did you need to make these changes?
This PR adds some error handling for the pair creation command. We are covering two edge cases:
* Pair initiators use the `pair` command without listing a username with '@'
  * e.g: (`Pairing!` or `Pairing with alan`)
* Pair initiators use an invalid @handle 
  * e.g: (`Pairing with @notarealperson`)

#### What changes did you make?
We added some validations for Swifty to DM the user and return before blowing up internally.

#### Obligatory gif

Obligatory Gif

### CODE REVIEW

File: routes/pairs.js [hunk 1]
```diff
@@ -36,8 +36,21 @@ router.post('/create', (req, res, next) => {
...
+
+    // if the initiator doesn't list a @username
+    if (!reqBodyText.includes('@')) {
+      return swifty.postToUser(initiatorName, 'I need a valid @username.');
+    };
+
+    var partnerID = reqBodyText.split('@')[1].split('>')[0];
     var partner = swifty.findUser(partnerID);
+
+    // if the initiator doesn't list an invalid @username
```
Comment (abrahamsangha) [inline routes/pairs.js]: I believe this should be 'if the initiator lists an invalid @username', unless I got lost in the double negative.

File: routes/pairs.js [hunk 2]
```diff
@@ -36,8 +36,21 @@ router.post('/create', (req, res, next) => {
...
+    if (!partner) {
+      return swifty.postToUser(initiatorName, 'I need a valid @username.');
+    };
```
Comment (craftninja) [inline routes/pairs.js]: Could we have both of these invalid cases in the same if statement?

```js
if (!partner || !reqBodyText.includes('@')) {
  return swifty.postToUser(initiatorName, 'I need a valid @username.');
};
```

And I'm not sure the abstraction of reqBodyText is super helpful... what can it be named to be more clear? Might be a fools errand...

File: routes/pairs.js (hunk 1 above)
Comment (alanbsmith) [inline routes/pairs.js]: yeah, the comment is incorrect. I changed my sentence midstream. good catch. It should read 'If the initiator uses an invalid @username'

File: routes/pairs.js (hunk 2 above)
Comment (alanbsmith) [inline routes/pairs.js]: Thought about that, but we have to check for a partner further down the line. You could pull the statement, 'I need a valid @username', into a separate method, but it feels like a little much to me.

Comment (craftninja) [inline routes/pairs.js]: Looked again, I do see what you mean. Getting the partner would blow up if there is not a proper @:user_id, correct?

Looks good :metal:

Comment (alanbsmith) [inline routes/pairs.js]: Correct. I'll change the comment Abraham mentioned above and merge.

**Pull request merged**